src/System.Private.CoreLib/shared/System/Array.cs

```diff
@@ -1569,14 +1569,15 @@ public static void Sort<T>(T[] array, int index, int length, System.Collections.
                     }
                 }
 #endif
-
                 if (items == null)
                 {
                     Sort<TKey>(keys, index, length, comparer);
                     return;
                 }
 
-                ArraySortHelper<TKey, TValue>.Default.Sort(keys, items, index, length, comparer);
+                var spanKeys = new Span<TKey>(ref Unsafe.As<byte, TKey>(ref keys.GetRawArrayData()), keys.Length);
+                var spanItems = new Span<TValue>(ref Unsafe.As<byte, TValue>(ref items!.GetRawArrayData()), items!.Length); // TODO-NULLABLE: https://github.com/dotnet/csharplang/issues/538
+                ArraySortHelper<TKey, TValue>.Default.Sort(spanKeys, spanItems, index, length, comparer!); // comparer may still be null here
```

Comment:
```suggestion
                ArraySortHelper<TKey, TValue>.Default.Sort(spanKeys, spanItems, index, length, comparer);
```

to match the original code?

Reply:
applied